### PR TITLE
Removes internal bleeding, adds arterial bleeding

### PR DESCRIPTION
**Pull request by ChungusGamer666**
## What Does This PR Do
Replaces internal bleeding with something better.
Mostly based on this baystation PR (https://github.com/Baystation12/Baystation12/pull/15529)

## Why It's Good For The Game
Internal bleeding SUCKS, and here is why:
1. Too easy to cause
2. Too hard to fix (requires an entire surgery)
3. Somewhat hard to diagnose (requires a full body scan to know the location)
4. Silent killer, when you notice it it's probably a tad late
5. All limbs have the same internal bleeding rate

What this does is replace it with torn arteries, and here is why they are better:
1. Torn arteries are easy to see, they can be seen on examine
2. Torn arteries are easy to hear, they squirt blood once every 10 seconds making a sound and spraying blood in a random direction
3. They are easy to fix, and only require a surgery of 3 steps (incision, fix o vein, cauterize)
4. Blunt weapons are very unlikely to cause arterial bleeding (you need at least a 20 damage blunt weapon)
5. Arterial bleeding is more or less severe depending on the limb

I am on a hurry and will elaborate more on this PR later if necessary.
Testing needs to be done.

I am also interested in porting tendons, but i am not sure if there is interest in such a thing and i would rather atomize this PR.

## Changelog
:cl:
del: Internal bleeding is gone.
add: Added arterial bleeding, a better version of internal bleeding.
/:cl:

### CODE REVIEW

**Comment by Adrer** [comment]
Don't see any reason to change internal bleeding, there are noticable signs to it, I don't see how shaving off 2 surgery steps off an already very easy surgery is required, and I personally at least prefer having the first steps of surgery standardized (Partially for Farie's surgery rework)

**Comment by ChungusGamer666** [comment]
> Don't see any reason to change internal bleeding, there are noticable signs to it, I don't see how shaving off 2 surgery steps off an already very easy surgery is required, and I personally at least prefer having the first steps of surgery standardized (Partially for Farie's surgery rework)

Those signs being a 5% chance every second to cough up blood assuming you have one internal bleeding limb, + 2.5% chance of vomiting blood.
To top it off the message for having internal bleeding is very geneirc and can be confused with other stuff such as organ damage.
Again, like i said, when you finally notice it, you'll probably be bled dry.

**Comment by hal9000PR** [comment]
I've never once had an issue with IB killing me before i noticed it. You cough up blood. The only other thing that makes you cough up blood is a ruptured lung, which makes you gasp aswell. Is it really that hard to use a health analyser on yourself? I don't see the point in these changes.

**Comment by ChungusGamer666** [comment]
> I've never once had an issue with IB killing me before i noticed it. You cough up blood. The only other thing that makes you cough up blood is a ruptured lung, which makes you gasp aswell. Is it really that hard to use a health analyser on yourself? I don't see the point in these changes.

Health analyzer does not tell you the location. A full body scan is required for that.

File: code/modules/surgery/organs/blood.dm [hunk 1]
```diff
@@ -67,19 +67,55 @@
...
+					to_chat(H, "<span class='danger'>You are hit by a spray of blood!</span>")
+					hit_mob = TRUE
+
+			if(hit_mob || !A.CanPass(src, sprayloc))
+				break
+
+		add_splatter_floor(sprayloc, FALSE, spraydir = spray_dir)
+		if(hit_mob)
+			return
+		sleep(1)
```
Comment (hal9000PR) [inline code/modules/surgery/organs/blood.dm:117]: What's the point in this `sleep` here?

Comment (ChungusGamer666) [reply code/modules/surgery/organs/blood.dm:117]: It's just to make the blood squirting do an animation instead of instantaneously splattering each tile.

**Comment by Hamurlik** [comment]
chungus

**Comment by hal9000PR** [comment]
> > I've never once had an issue with IB killing me before i noticed it. You cough up blood. The only other thing that makes you cough up blood is a ruptured lung, which makes you gasp aswell. Is it really that hard to use a health analyser on yourself? I don't see the point in these changes.
> 
> Health analyzer does not tell you the location. A full body scan is required for that.

Why do you need to know the location? The only purpose of that is surgery and you cannot do surgery on yourself so you need another person... Who is normally in medbay... With a body scanner. 



So let's go over your points of why internal bleeding sucks 

"too easy to cause" you haven't listed how this has been changed. Now sharp weapons are more likely to cause IB, which is an alright change. I feel that the internal damage system needs a major overhaul in how its calculated and caused.

"Too hard to fix" so we have gone from a 5 step surgery to a three step surgery? This is a tiny change which basically makes no difference. What's the point in dropping two surgery steps? It doesn't make it any easier to get fixed, cos you still need surgery which is the hard part.

"Too hard to diagnose" coughing up blood and not randomly gasping? You got IB. I've already been over the location thing

"Silent killer" if you don't notice yourself coughing up blood, you sorta deserve dying after 20 minutes or so.

"All limbs have the same bleed rate" okay I can see this is a decent point, but how much is this really needed?

This is a nerf to any simple mob combat as simple mobs are less likely to deal IB as bites arent sharp. This means miners are stronger and get their gamer loot slightly easier. There are many balance applications to this that aren't ideal. Such as buffing slime people. If the only type of internal damage that affects them is now worse, they are stronger.

**Comment by Wojz** [comment]
My issue with this is that it tries to oversimplify medical even more. You're not supposed to know where the IB is, that is by design. It's already the easiest surgery that exists. If you run around with an IB long enough that you die but somehow do not notice your character coughing up blood - which can only mean IB or a ruptured lung, another emergency, even worse one in fact - that's actually unironically a skill issue. Pay attention to the chat long and it won't happen.

**Comment by Gatchapod** [comment]
You don't even need a body scanner.
Internal bleeding surgery option appears only if a targeted body part has an internal bleeding, so you can just keep trying different parts until you find it. It's cheesy, but works.

With one IB, you start getting red warnings regarding your blood level below 80%, which is still safe. Actual oxyloss damage starts building up around 60%, I believe, and it's still extremely slow.
If you have multiple IBs you'll get periodically stunned, which is an instant giveaway.

Why would they be visible via examine? We already have that and it's called external bleeding. It's bleeding that's meant to be visible with a naked eye and generally an open wound. Internal bleeding is meant to be dangerous and, well, hidden. That's its entire gimmick. Being harder to identify.

Finally, I don't see anything in your code addressing the issue of all limbs having the same IB/arterial bleeding rate, which is the only thing that might actually be worth considering.

Overall, the entire PR gives a vibe of "i ded nerf pls".

**Comment by ChungusGamer666** [comment]
> You don't even need a body scanner.
> Internal bleeding surgery option appears only if a targeted body part has an internal bleeding, so you can just keep trying different parts until you find it. It's cheesy, but works.
> 
> With one IB, you start getting red warnings regarding your blood level below 80%, which is still safe. Actual oxyloss damage starts building up around 60%, I believe, and it's still extremely slow.
> If you have multiple IBs you'll get periodically stunned, which is an instant giveaway.
> 
> Why would they be visible via examine? We already have that and it's called external bleeding. It's bleeding that's meant to be visible with a naked eye and generally an open wound. Internal bleeding is meant to be dangerous and, well, hidden. That's its entire gimmick. Being harder to identify.
> 
> Finally, I don't see anything in your code addressing the issue of all limbs having the same IB/arterial bleeding rate, which is the only thing that might actually be worth considering.
> 
> Overall, the entire PR gives a vibe of "i ded nerf pls".

sorry buddy i don't play on paradise

Anyways if you read the code i added a variable called arterial_bleed_severity, the head artery (carotid) being the worst bleeder.

**Comment by ChungusGamer666** [comment]
> My issue with this is that it tries to oversimplify medical even more. You're not supposed to know where the IB is, that is by design. It's already the easiest surgery that exists. If you run around with an IB long enough that you die but somehow do not notice your character coughing up blood - which can only mean IB or a ruptured lung, another emergency, even worse one in fact - that's actually unironically a skill issue. Pay attention to the chat long and it won't happen.

This doesn't simplify medical. Arterial bleeding kills way quicker than internal bleeding, the counter being the fact it happens less often and it is easier to fix.

**Comment by ChungusGamer666** [comment]
Response was too lukewarm for my tastes, i'll just close the PR.
If some hero wants to pick up from my work and fight this uphill battle, go ahead.

**Comment by LightFire53** [comment]
To be honest, I don't see why this couldn't have been added in conjunction with internal bleeding. In reality, a patient could realistically have both, so why not in spess

**Comment by ChungusGamer666** [comment]
> To be honest, I don't see why this couldn't have been added in conjunction with internal bleeding. In reality, a patient could realistically have both, so why not in spess

Considering precendent for medical simplification here, i doubt the maintainers would want a sidegrade to internal bleeding.

**Comment by Spacemanspark** [comment]
>This doesn't simplify medical.

I would like to point out that it's a little odd you say this... but everything you say in the "why is this good" section says it's "easier".

I would be all for this being added alongside internal bleeding, however... but I don't play organics much these days, so my opinion doesn't carry much weight here.

**Comment by ChungusGamer666** [comment]
> > This doesn't simplify medical.
> 
> I would like to point out that it's a little odd you say this... but everything you say in the "why is this good" section says it's "easier".
> 
> I would be all for this being added alongside internal bleeding, however... but I don't play organics much these days, so my opinion doesn't carry much weight here.

I mean, this simplifies medical, but doesn't make it any easier. That is what i meant all along.

**Pull request closed without merging**